src/main/java/uk/gov/hmcts/reform/mi/miextractionservice/component/impl/CoreCaseDataExportBlobDataComponentImpl.java

```diff
@@ -0,0 +1,152 @@
+import uk.gov.hmcts.reform.mi.miextractionservice.exception.ExportException;
+import uk.gov.hmcts.reform.mi.miextractionservice.model.OutputCoreCaseData;
+import uk.gov.hmcts.reform.mi.miextractionservice.util.DateTimeUtil;
+import uk.gov.hmcts.reform.mi.miextractionservice.util.ReaderUtil;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Component
+public class CoreCaseDataExportBlobDataComponentImpl implements ExportBlobDataComponent {
+
+    private static final String CCD_OUTPUT_CONTAINER_NAME = "ccd";
+    private static final String NAME_DELIMITER = "-";
+    private static final String CCD_DATA_CONTAINER_PREFIX = CCD_OUTPUT_CONTAINER_NAME + NAME_DELIMITER;
+    private static final String WORKING_FILE_NAME = "CCD_EXTRACT.csv";
+    private static final String WORKING_ARCHIVE = "CCD_EXTRACT.zip";
+
+    @Autowired
+    private BlobDownloadComponent<byte[]> blobDownloadComponent;
+
+    @Autowired
+    private DataParserComponent<CoreCaseData> dataParserComponent;
+
+    @Autowired
+    private CoreCaseDataFormatterComponent<OutputCoreCaseData> coreCaseDataFormatterComponent;
+
+    @Autowired
+    private CsvWriterComponent<OutputCoreCaseData> csvWriterComponent;
+
+    @Autowired
+    private  EncryptArchiveComponent encryptArchiveComponent;
+
+    @Autowired
+    private GenerateBlobUrlComponent generateBlobUrlComponent;
+
+    @Autowired
+    private ReaderUtil readerUtil;
+
+    @Autowired
+    private DateTimeUtil dateTimeUtil;
+
+    @SuppressWarnings("PMD.LawOfDemeter")
+    @Override
+    public String exportBlobsAndReturnUrl(BlobServiceClient sourceBlobServiceClient,
+                                          BlobServiceClient targetBlobServiceClient,
+                                          OffsetDateTime fromDate,
+                                          OffsetDateTime toDate) {
+
+        List<OutputCoreCaseData> outputData = new ArrayList<>();
+
+        List<String> blobNameIndexes = getYearMonthIndexes(fromDate, toDate);
+
+        for (BlobContainerItem blobContainerItem : sourceBlobServiceClient.listBlobContainers()) {
+            if (blobContainerItem.getName().startsWith(CCD_DATA_CONTAINER_PREFIX)) {
+                for (BlobItem blobItem : sourceBlobServiceClient.getBlobContainerClient(blobContainerItem.getName()).listBlobs()) {
+                    if (blobNameIndexes.parallelStream().anyMatch(blobItem.getName()::contains)) {
+
+                        byte[] blobData = blobDownloadComponent
+                            .downloadBlob(sourceBlobServiceClient, blobContainerItem.getName(), blobItem.getName());
+
+                        List<String> stringData = readerUtil.readBytesAsStrings(blobData);
+
+                        outputData.addAll(convertDataStringsToOutput(stringData, fromDate, toDate));
```

Comment:
the method name is misleading, as it missing the filter part
probably is cleaner to this in two steps using streams
stringData.stream().filter(fromDate, toDate).map(toOutputCoreCaseData).collect()